### PR TITLE
PR4

### PR DESCRIPTION
**Pull request by jokano12**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

- [ ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 1]
```diff
@@ -2364,4514 +3373,5821 @@
...
+        ],
+        "x-ms-visibility": "advanced"
+      }
+    }
+  },
+  "definitions": {
+    "CommitteeNomineeData": {
+      "type": "object",
+      "properties": {
+        "id": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:5699]: Please fix the description and title

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 2]
```diff
@@ -770,65 +192,112 @@
...
-                    true,
-                    false
-                  ]
-                },
-                "address": {
-                  "type": "string",
-                  "description": "Address.",
-                  "title": "Address"
-                },
                 "type": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:194]: Value is required

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 3]
```diff
@@ -770,65 +192,112 @@
...
                   "type": "string",
                   "description": "Type.",
                   "title": "Type"
                 },
-                "showInDirectory": {
-                  "type": "boolean",
-                  "description": "Show In Directory",
-                  "title": "Show In Directory",
+                "url": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:210]: This one is also required

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 4]
```diff
@@ -770,65 +192,112 @@
...
+            "x-ms-visibility": "internal"
+          },
+          {
+            "name": "body",
+            "in": "body",
+            "required": false,
+            "schema": {
+              "type": "object",
+              "properties": {
+                "type": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:264]: This one is also required

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 5]
```diff
@@ -770,65 +192,112 @@
...
+                    "Facebook",
+                    "Twitter",
+                    "Website",
+                    "Domain",
+                    "Other",
+                    "Instagram",
+                    "YouTube"
                   ]
+                },
+                "url": {
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:280]: This one is also required

File: certified-connectors/Impexium/apiDefinition.swagger.json (hunk 2 above)
Comment (lordlydon) [reply certified-connectors/Impexium/apiDefinition.swagger.json:194]: Also, please add enum options here too

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 6]
```diff
@@ -930,93 +396,80 @@
             "default": "application/json",
             "x-ms-visibility": "internal"
           }
-        ]
+        ],
+        "summary": "Get Organization relationships",
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please change to `Get Organization's Relationships`, to match the format of the Individual's equivalent action

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 7]
```diff
@@ -930,93 +396,80 @@
...
-          },
-          {
-            "name": "Content-Type",
-            "in": "header",
-            "required": true,
-            "type": "string",
-            "default": "application/json",
-            "x-ms-visibility": "internal"
+            "schema": {
+              "$ref": "#/definitions/AddressSaveData"
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json:460]: I missed this one last time, can you add the enum options for the address type: Home, Work and Other.

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 8]
```diff
@@ -930,93 +396,80 @@
...
       "post": {
-        "tags": ["Individuals"],
+        "tags": [
+          "Individuals"
+        ],
         "responses": {
           "200": {
-            "description": "Success",
+            "description": "default",
             "schema": {}
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please check the response definition at http://abc.mpxdemo.com/api/help/Api/POST-api-v1-Individuals-idOrRecordNumber-CustomFieldsList

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 9]
```diff
@@ -1130,34 +559,36 @@
...
           "200": {
-            "description": "200",
+            "description": "Success",
             "schema": {}
           }
         },
-        "summary": "Add Relationship to Individual",
-        "description": "Add Relationship to Individual.",
-        "operationId": "Add-relationship-to-individual",
+        "summary": "Update Custom Field Value",
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Please change to `Update Individual's Custom Field Value` as there is a different endpoint to update values for Organizations

**Review by lordlydon — CHANGES_REQUESTED** [review]

File: certified-connectors/Impexium/apiDefinition.swagger.json (hunk 7 above)
Comment (jokano12) [reply certified-connectors/Impexium/apiDefinition.swagger.json:460]: I can add the enum option to the ref "AddressSaveData", will this work? Or does this option not apply to all actions with "addresssavedata." If the latter is the case, I can remove the reference and add the schema with the enum in it.

Comment (lordlydon) [reply certified-connectors/Impexium/apiDefinition.swagger.json:460]: Yes, adding it to the ref would work

File: certified-connectors/Impexium/apiDefinition.swagger.json (hunk 8 above)
Comment (jokano12) [reply certified-connectors/Impexium/apiDefinition.swagger.json]: When I add the schema, it causes the flow to fail and result in the following. However, I remove the schema and it works. And note that the schema that is missing is the response schema, not the body schema which is referenced in the error.
Screenshot 2021-11-25 185937
Screenshot 2021-11-25 185919

Comment (lordlydon) [reply certified-connectors/Impexium/apiDefinition.swagger.json]: There is an error with the swagger definition, the response is an array of `CustomFieldResultData` http://abc.mpxdemo.com/api/help/ResourceModel?modelName=CustomFieldResultData

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 10]
```diff
@@ -9271,7 +9284,12 @@
         "type": {
           "type": "string",
           "description": "Type.",
-          "title": "Type"
+          "title": "Type",
+          "enum": [
+            "Home",
+            "Work",
+            "Other"
+            [
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Added wrong bracket here

**Review by lordlydon — CHANGES_REQUESTED** [review]

File: certified-connectors/Impexium/apiDefinition.swagger.json (hunk 8 above)
Comment (jokano12) [reply certified-connectors/Impexium/apiDefinition.swagger.json]: Screenshot 2021-11-28 205754
Adding the response definition causes paconn to run an error with it. In addition, the flow doesnt work. I removed the response definition and paconn succeeded and the flow works.

Comment (lordlydon) [reply certified-connectors/Impexium/apiDefinition.swagger.json]: I don't get any errors adding it like this, can you check it again?
```
"responses": {
  "200": {
    "description": "default",
    "schema": {
      "type": "array",
      "items":{
        "$ref": "#/definitions/CustomFieldResultData"
      }
    }
  }
},
```

Comment (jokano12) [reply certified-connectors/Impexium/apiDefinition.swagger.json]: I got the flow working, but I am still gettting the paconn error. 
Screenshot 2021-11-30 145854
You arent getting the error when run through paconn? What does your definition for customfieldresultdata look like?

Comment (lordlydon) [reply certified-connectors/Impexium/apiDefinition.swagger.json]: I didn't make any changes to the CustomFieldResultData definition:
```
"CustomFieldResultData": {
  "type": "object",
  "properties": {
    "status": {
      "type": "string",
      "description": "Status.",
      "title": "Status"
    },
    "message": {
      "type": "string",
      "description": "Message.",
      "title": "Message"
    },
    "customFieldName": {
      "type": "string",
      "description": "Custom Field Name.",
      "title": "Custom Field Name"
    }
  }
},
```

Comment (jokano12) [reply certified-connectors/Impexium/apiDefinition.swagger.json]: My paconn still doesnt validate it. Any suggestions?

Comment (lordlydon) [reply certified-connectors/Impexium/apiDefinition.swagger.json]: Can you commit the changes so I can check?

Comment (jokano12) [reply certified-connectors/Impexium/apiDefinition.swagger.json]: Im not sure whats different but, I checked paconn again with the file I committed to this PR and it validated this time.

**Review by lordlydon — COMMENTED** [review]
Looks good.
@RjImpx, we should wait until the previous actions are merged into the Microsoft repository before merging this pull request.

**Comment by lordlydon** [comment]
@jokano12, we merged the other PR and now there are some conflicts for this one. Can you fix these?

**Review by lordlydon — APPROVED** [review]

File: certified-connectors/Impexium/apiDefinition.swagger.json [hunk 11]
```diff
@@ -21,30 +21,66 @@
   "produces": [
     "application/json"
   ],
-  "paths": {
-    "/api/v1/Individuals/{ID or Record Number}/Phones": {
-      "post": {
+  "paths":
```
Comment (lordlydon) [inline certified-connectors/Impexium/apiDefinition.swagger.json]: Missing `{`

**Review by lordlydon — CHANGES_REQUESTED** [review]

**Review by lordlydon — APPROVED** [review]

**Pull request merged**